File: taskmanager/src/main/java/com/tasksphere/taskmanager/application/service/TaskService.java
package com.tasksphere.taskmanager.application.service;

import com.tasksphere.taskmanager.application.dto.task.CreateTaskRequest;
import com.tasksphere.taskmanager.application.dto.task.TaskResponse;
import com.tasksphere.taskmanager.application.dto.task.UpdateTaskStatusRequest;
import com.tasksphere.taskmanager.application.dto.task.UpdateTaskRequest;
import com.tasksphere.taskmanager.application.dto.task.TaskSearchRequest;
import com.tasksphere.taskmanager.application.dto.tag.TagResponse;
import com.tasksphere.taskmanager.application.dto.statistics.TaskStatisticsResponse;
import com.tasksphere.taskmanager.application.dto.comment.CreateCommentRequest;
import com.tasksphere.taskmanager.application.dto.comment.CommentResponse;

import java.util.List;

public interface TaskService {
    List<TaskResponse> getTasks();
    TaskResponse createTask(CreateTaskRequest request);
    List<TaskResponse> getMyTasks();
    List<TaskResponse> getCreatedTasks();
    TaskResponse updateTaskStatus(Long taskId, UpdateTaskStatusRequest request);
    TaskResponse updateTask(Long taskId, UpdateTaskRequest request);
    void deleteTask(Long taskId);
    List<TaskResponse> searchTasks(TaskSearchRequest request);
    void addTagToTask(Long taskId, Long tagId);
    void removeTagFromTask(Long taskId, Long tagId);
    List<TagResponse> getTaskTags(Long taskId);
    void addCollaborator(Long taskId, Long userId);
    void removeCollaborator(Long taskId, Long userId);
    List<TaskResponse> getCollaborativeTasks();
    TaskStatisticsResponse getStatistics();
    CommentResponse addComment(Long taskId, CreateCommentRequest request);
    List<CommentResponse> getTaskComments(Long taskId);
} 

File: taskmanager-frontend/src/services/task.service.ts
import { api } from './api'
import { Task, CreateTaskRequest, UpdateTaskRequest, UpdateTaskStatusRequest, Comment } from '@/types/task'

interface AddCommentRequest {
  content: string;
}

class TaskService {
  async getTasks(): Promise<Task[]> {
    const response = await api.get<Task[]>('/tasks')
    return response.data
  }

  async createTask(data: CreateTaskRequest): Promise<Task> {
    const response = await api.post<Task>('/tasks', data)
    return response.data
  }

  async updateTask(taskId: number, data: UpdateTaskRequest): Promise<Task> {
    const response = await api.put<Task>(`/tasks/${taskId}`, data)
    return response.data
  }

  async updateTaskStatus(taskId: number, data: UpdateTaskStatusRequest): Promise<Task> {
    const response = await api.patch<Task>(`/tasks/${taskId}/status`, data)
    return response.data
  }

  async deleteTask(taskId: number): Promise<void> {
    await api.delete(`/tasks/${taskId}`)
  }

  async addComment(taskId: number, data: AddCommentRequest): Promise<Comment> {
    const response = await api.post<Comment>(`/tasks/${taskId}/comments`, data)
    return response.data
  }

  async addCollaborator(taskId: number, userId: number): Promise<void> {
    await api.put(`/tasks/${taskId}/collaborators/${userId}`)
  }

  async removeCollaborator(taskId: number, userId: number): Promise<void> {
    await api.delete(`/tasks/${taskId}/collaborators/${userId}`)
  }
}

export const taskService = new TaskService() 